test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java

```diff
@@ -658,15 +658,17 @@ protected IndexShard newShard(
                 remoteStorePressureService = new RemoteStorePressureService(clusterService, indexSettings.getSettings());
                 BlobStoreRepository repo = createRepository(remotePath);
                 when(mockRepoSvc.repository(any())).thenAnswer(invocationOnMock -> repo);
+            } else {
+                remoteStorePressureService = null;
```

Comment:
curious to find out the reason for the change. The value is null by default and we set it to non null value if remote enabled.

Reply:
I don't recall it fully but this was to address a code analysis-flagged error where it required this variable to be effectively final in all cases or something. However, the dependency on pressure service is now removed in the latest revision due to the tracker factory change.